### PR TITLE
Update database password

### PR DESCRIPTION
**Pull request by neelxie**
# Description

This task adds functionality for updating/changing both PostgreSQL and MySQL database passwords.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Trello Ticket ID

https://trello.com/c/7dfobMCU

## How Can This Been Tested?

Pull this branch and run it locally, click on a particular database and click the button to update its password, in the modal fill in the  new password and click update.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

# Screenshots
Screenshot from 2021-07-21 08-53-23

Screenshot from 2021-07-20 04-16-02
Screenshot from 2021-07-20 04-56-36

### CODE REVIEW

File: src/components/DBSettingsPage/index.jsx [hunk 1]
```diff
@@ -348,6 +439,69 @@ class DBSettingsPage extends React.Component {
...
+                      showModal={openUpdateModal}
+                      onClickAway={this.hideUpdateModal}
+                    >
+                      <div className="DeleteDatabaseModel">
+                        <div className="DeleteProjectModalUpperSection">
+                          <div className="ModalFormHeading">
+                            <h2>Update database password</h2>
+                          </div>
+                          <div className="InnerModalDescription">
+                            <BlackInputText
```
Comment (fel-w) [inline src/components/DBSettingsPage/index.jsx:464]: How is the idea of adding a **Confirm Password** input as well?

Comment (neelxie) [reply src/components/DBSettingsPage/index.jsx:464]: @fel-w kindly elaborate on this

Comment (fel-w) [reply src/components/DBSettingsPage/index.jsx:464]: @neelxie, something like this:
image

Comment (neelxie) [reply src/components/DBSettingsPage/index.jsx:471]: @fel-w I have worked on the feedback
Screenshot from 2021-07-21 18-26-31

**Review by fel-w — APPROVED** [review]

**Review by rhodinemma — APPROVED** [review]

**Review by StephenAraka — COMMENTED** [review]
image

@neelxie for this modal, is it possible to [vertically] space out the elements inside it? I think they are a little too close to each other. Especially the heading and the input field

**Comment by MutegekiHenry** [comment]
> image
> 
> @neelxie for this modal, is it possible to [vertically] space out the elements inside it? I think they are a little too close to each other. Especially the heading and the input field

**Review by StephenAraka — APPROVED** [review]

**Comment by MutegekiHenry** [comment]
> image
> 
> @neelxie for this modal, is it possible to [vertically] space out the elements inside it? I think they are a little too close to each other. Especially the heading and the input field

@neelxie on this same model I have a few checks that we need to have in place.

- For starters, the error message is a bit misleading, the maximum number of characters is 32 so it should be at-most not at least
- Then there are some characters we don't allow in the password that causes it to fail at the host. Currently, we only allow all letters and numbers, and these selected special chars `!#%&()+,-:;<=>?@[]^_{|}~` anything outside this should be rejected.

**Review by khalifan-kfan — DISMISSED** [review]

**Comment by kawerewagaba** [comment]
@neelxie let's resolve the conflicts for starters

**Comment by neelxie** [comment]
> @neelxie let's resolve the conflicts for starters

They have been resolved @kawerewagaba

**Review by MutegekiHenry — APPROVED** [review]
Looks good

**Pull request merged**